specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/virtualclusters.json

```diff
@@ -0,0 +1,453 @@
+          }
+        },
+        "x-ms-examples": {
+          "Get virtual cluster": {
+            "$ref": "./examples/VirtualClusterGet.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "VirtualClusters"
+        ],
+        "description": "Deletes a virtual cluster.",
+        "operationId": "VirtualClusters_Delete",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/VirtualClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully deleted the virtual cluster."
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 SubscriptionNotFound - The requested subscription was not found.\n\n * 400 ManagementServiceFeatureDisabled - User attempted to use a feature which is disabled.\n\n * 404 ResourceDoesNotExist - Resource with the name '{0}' does not exist. To continue, specify a valid resource name.\n\n * 404 VirtualClusterNotInSubscriptionResourceGroup - specified virtual cluster does not exist in the specified resource group and subscription.\n\n * 404 SubscriptionDoesNotHaveVirtualCluster - The requested virtual cluster was not found\n\n * 404 OperationIdNotFound - The operation with Id does not exist.\n\n * 404 ResourceNotFound - The requested resource was not found.\n\n * 409 ConflictingVirtualClusterOperation - An operation is currently in progress for the virtual cluster.\n\n * 409 OperationCancelled - The operation has been cancelled by user.\n\n * 409 OperationInterrupted - The operation on the resource could not be completed because it was interrupted by another operation on the same resource.\n\n * 429 ConflictingSubscriptionOperation - An operation is currently in progress for the subscription.\n\n * 500 OperationTimedOut - The operation timed out and automatically rolled back. Please retry the operation."
+          },
+          "202": {
+            "description": "Deleting the virtual cluster is in progress."
+          },
+          "204": {
+            "description": "The specified virtual cluster does not exist."
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Delete virtual cluster": {
+            "$ref": "./examples/VirtualClusterDelete.json"
+          }
+        }
+      },
+      "patch": {
```

Comment:
How is the virtualClusters resource created? Given the lack of a put?

Reply:
We don't have a Put for virtualCluster resource since it is created indirectly, when managedInstance is created (first managedInstance within a subnet). It serves as a container for multiple managedInstance resources, and is directly mapped to one of our internal resources.